### ISSUE TITLE
could Cartodb.js be more modular?

### ISSUE THREAD

**Issue by standardpixel** [CartoDB/cartodb.js#438]
Hello,

I am trying to use cartodb in a couple of applications. In both cases it is a small part of a bigger architecture and design. Here are the scenarios:

1. 
   * scenario: I need to display a cartodb layer over an existing map. 
   * problem: The cartodb createVis method overides the styles and behavior of the map. My styled zoom controls backgrounds change. All that I really need here is a layer template. Instead it makes a lot of assumptions about my map configuration

2.
   * scenario: I am using cartodb as the backend api for a React.JS app which uses NPM as a front-end package manager. 
   * problem: cartodb.js exists in NPM but requires a build step. It does not work as a module.

The entire library took 500ms to load also. It seems to have a lot of dependencies which I did not need just to use the query method. 

Basically, can simple client and tilelayer building modules be abstracted without dependence on JQuery or Backbone. Ones which can be installed using NPM?

Thanks!

**Comment by javisantana** [comment]
maybe what you are looking for is cartodb.core  (http://libs.cartocdn.com/cartodb.js/v3/3.13/cartodb.core.js) it basically provides:

- sql api access
- access to tile XYZ urls for a sql + cartocss (utf grid urls too)
- preview api 

what it does NOT provide:

- createVis
- createLayer
- anything related with interactivity (infowindows...) 
- basic interaction, you would need to use leaflet.wax to use utfgrids

http://docs.cartodb.com/cartodb-platform/cartodb-js.html#core-api-functionality

Is that useful or you'd need something more powerful?

**Comment by standardpixel** [comment]
That looks like it will work. Thanks!

It would be great if this could be packaged so that it could be loaded through NPM without need for a build. Something like `npm install cartodb-core`.

Thanks again!
